h5c/vic/src/vic-webapp/src/app/shared/components/vch-storage/vch-storage.component.ts

```diff
@@ -0,0 +1,207 @@
+import {VchComponentBase} from '../vch-component-base';
+import {Component, Input, OnInit} from '@angular/core';
+import {CreateVchWizardService} from '../../../create-vch-wizard/create-vch-wizard.service';
+import {ConfigureVchService, SelectedComputeResourceInfo} from '../../../configure/configure-vch.service';
+import {GlobalsService} from '../../globals.service';
+import {FormArray, FormBuilder, Validators} from '@angular/forms';
+import {Observable} from 'rxjs/Observable';
+import {ImageSizeUnit, VchUiStorage, VchUiStorageVolumeStore} from '../../../interfaces/vch';
+import {numberPattern, supportedCharsPattern} from '../../utils/validators';
+import {noBlankSpaces} from '../../utils';
+import {ServerInfo} from '../../vSphereClientSdkTypes';
+import {ComputeResource} from '../../../interfaces/compute.resource';
+
+@Component({
+  selector: 'vic-vch-storage',
+  templateUrl: './vch-storage.component.html',
+  styleUrls: ['./vch-storage.component.scss']
+})
+export class VchStorageComponent extends VchComponentBase implements OnInit {
+
+  @Input() set resourceObjRef(value) {
+    if (typeof value !== 'undefined') {
+      this.loadDatastore(value);
+    }
+  }
+
+  model: VchUiStorage;
+  initialModel: VchUiStorage = {
+    baseImageSize: '8',
+    baseImageSizeUnit: 'GiB',
```

Comment:
It sounds weird but it was like this before my changes... I didn't wanted to changed, just in case to avoid introduce errors. We can review it later if you want.

Reply:
Yeah, let's follow up on this to be safe.